src/bin/fish_indent.rs

```diff
@@ -162,39 +162,36 @@ impl<'source, 'ast> PrettyPrinter<'source, 'ast> {
 
     // Return the gap ranges from our ast.
     fn compute_gaps(&self) -> Vec<SourceRange> {
-        let range_compare = |r1: SourceRange, r2: SourceRange| {
-            (r1.start(), r1.length()).cmp(&(r2.start(), r2.length()))
-        };
-        // Collect the token ranges into a list.
-        let mut tok_ranges = vec![];
-        for node in Traversal::new(self.ast.top()) {
+        let mut tok_ranges: Vec<SourceRange> = Traversal::new(self.ast.top())
+        .filter_map(|node| {
             if node.category() == Category::leaf {
-                let r = node.source_range();
-                if r.length() > 0 {
-                    tok_ranges.push(r);
+                let range = node.source_range();
+                if range.length() > 0 {
+                    Some(range)
+                } else {
+                    None
                 }
+            } else {
+                None
             }
+        })
+        .collect();
```

Comment:
A little idea in regards to code simplification: using some shorter `.map()`s and `.filter()`s instead of a general `.filter_map()`, those dangling `else { None }`s could be avoided

```rust
        let mut tok_ranges: Vec<SourceRange> = Traversal::new(self.ast.top())
            .filter(|node| node.category() == Category::leaf)
            .map(|leaf| leaf.source_range())
            .filter(|range| range.length() > 0)
            .collect();
```

I find this a bit easier to follow than the nested `if`s :)

(BTW, i cannot vouch for the correctness of these changes; this is just a drive-by comment.)